platform.py

```diff
@@ -110,6 +110,27 @@ def _add_default_debug_tools(self, board):
                 },
                 "onboard": True
             }
+        if debug.get("avr_debugger", ""):
+            speed = debug["avr_debugger"]["speed"]
```

Comment:
It seems like `115200` is the default value for this tool, Does it make sense to remove this `speed` field from board manifests, and move it to this file, something like:
```
"set serial baud {0}".format(debug["avr_debugger"].get("speed", 115200)),
```

Reply:
Certainly, the default for any board with a f_clock of 8MHz or more is 115200 as you can see [here](https://github.com/jdolinay/avr_debug/blob/4532f6cae98b319d16f289e3bf8e73634374a462/avr8-stub/avr8-stub.c#L142). However, a baud rate of 115200 is too high for a f_clock less than 8MHz, which is a pretty common case on low power or battery-powered systems. Then, the idea is to use that field to take care of such cases.

Now, even if the manifest doesn't have to explicitly say what speed to use, it still has to tell if the board is compatible with the avr_debugger stub as not all the avr MCUs are supported yet.

For what you propose the debug entry would have to be changed to something like this
~~~
    "debug": {
        "simavr_target": "atmega328p",
        "avr_debugger": {
            "supported": true
        }
    }
~~~
So the 'debug["avr_debugger"].get("speed", 115200)'  uses the speed entry if it exists or defaults to 115200.

Other option would be to create a 'debug_speed' variable for the 'platformio.ini', and use the value given by the user of a default which could be 115200, then the  manifest file would be reduced to:
~~~
    "debug": {
        "simavr_target": "atmega328p",
        "avr_debugger": true
    }
~~~

The last option would be to remove the avr_debugger parent like this:
~~~
    "debug": {
        "simavr_target": "atmega328p",
        "avr_debugger_speed": 115200
    }
~~~
In this case, the existence of the `avr_debugger_speed` entry would have the double purpose of telling if the tool is supported and the speed that should be used.

Even better, we could use the previous option which already gives us a default speed and the `debug_speed` field of the second option, so the user can use custom values for the baud rate.

I like this last one because the avr_debugger tool is able to use a custom speed by defining the [AVR8_USER_BAUDRATE](https://github.com/jdolinay/avr_debug/blob/4532f6cae98b319d16f289e3bf8e73634374a462/avr8-stub/avr8-stub.h#L25) macro, which automatically covers the case of requiring a lower baud rate.

What do you think?